### ISSUE TITLE
signMessage with Ledger is not working

### ISSUE THREAD

**Issue by neverything** [phantom-labs/sandbox#14]
* Ledger nano s, latest firmware 2.0.0
* Solana (SOL) app installed, version 1.0.6
* Google Chrome: Version 95.0.4638.54 (Official Build) (arm64)
* Blind sign is turned on
* Things in https://help.phantom.app/hc/en-us/articles/4406388670483-How-to-use-your-Ledger-Nano-hardware-wallet tried
* Ad blockers and content blockers disabled

Same error persists on the following sites:

* https://codesandbox.io/s/github/phantom-labs/sandbox?file=/src/App.tsx
* https://www.solanium.io/
* A local test environment I'm working on

Error message after clicking "Approve" on Signature Request:

```
Phantom - RPC Error: User rejected the request. 
{code: 4001, message: 'User rejected the request.'}
```

**Comment by fragosti** [comment]
Hi @neverything, this feature doesn't work on Ledger unfortunately. We are working with the Solana foundation and Ledger to add it to the Ledger firmware, but it could take a while.

**Comment by neverything** [comment]
Hey @fragosti thank you so much for your reply and that you are working on this with Solana and Ledger. I'll figure out a workaround to not use the signMessage feature for now.

**Comment by ericecchou** [comment]
Hey , guys any update on signMessage function?

**Comment by tiditidi** [comment]
Im still having this issue on Magic Eden and Opensea. Any updates?

**Comment by ChewingGlass** [comment]
@fragosti any updates, why did this issue get closed?

**Comment by bfriel** [comment]
> @fragosti any updates, why did this issue get closed?

This has been on Solana's roadmap for a while, but it got deprioritized earlier this year due to network congestion issues. They are aware of its importance and are hoping to have it done ASAP. There is not much the community can do here. Per Trent Nelson:

> FWIW I'm not generally interested in a community contribution to the Ledger App for this issue. There's some refactoring that should take place first and it will take longer to communicate the requirements to an unfamiliar contributor than to implement it myself.

**Comment by mayurbhandary** [comment]
Are there any updates or workarounds for this?

**Comment by fragosti** [comment]
Solana recently added this ability to the Ledger app https://github.com/solana-labs/solana/issues/21366

It's not clear we'll add it outside of our multi-chain re-write (https://phantom.app/waitlist). Should come out in Jan.

**Comment by matusvla** [comment]
@fragosti What is the status of this? Is the Jan date still standing?

**Comment by fragosti** [comment]
It is looking more like February now. As soon as multi-chain is out
basically.

Best,
Francesco
CTO & Co-founder, Phantom

**Comment by MrOpperman** [comment]
@fragosti thanks for the update, do you have any more news? Multi-chain is out, but doesn't seem like sign message is yet?

**Comment by eliahilse** [comment]
anything new @fragosti ?  On the multichain beta and still doesn't work :/

**Comment by bfriel** [comment]
> anything new @fragosti ? On the multichain beta and still doesn't work :/

We were advised by Solana that the Ledger message signing firmware in `1.3.0` is actually incomplete. They recommended that we wait for the final release which should land in `1.3.2` ([source](https://github.com/solana-labs/solana/compare/master...t-nelson:solana:ocmsf)).

We will be ready to integrate as soon as that firmware version lands.

**Comment by c4b4d4** [comment]
Seems to be happening soon? Been waiting for this for a long time 🚀

**Comment by conky-dev** [comment]
Please fix ser.

**Comment by GrimLothar** [comment]
Hey @neverything can you reopen this issue since it's not fixed yet? Unless there is another one tracking this in this repo....

**Comment by n526** [comment]
> > anything new @fragosti ? On the multichain beta and still doesn't work :/
> 
> We were advised by Solana that the Ledger message signing firmware in `1.3.0` is actually incomplete. They recommended that we wait for the final release which should land in `1.3.2` ([source](https://github.com/solana-labs/solana/compare/master...t-nelson:solana:ocmsf)).
> 
> We will be ready to integrate as soon as that firmware version lands.

We are already on v1.3.1 for a long time, when will v1.3.2 come? Is there any date?

**Comment by fakenine** [comment]
Hello! I just installed firmware update 1.4.1 and the signature is still not working.

**Comment by EternityTech** [comment]
cmon, we need this

**Comment by n526** [comment]
> Hello! I just installed firmware update 1.4.1 and the signature is still not working.

The version number given, is related to the Solana App which you install on your ledger and not the firmware version of ledger device. TMK it’s still on 1.3.1. dunno but I feel this is still going to take some more months or even longer till we see the requested update

**Comment by fakenine** [comment]
@bfriel Anything new from Solana Labs?
Feels like eternity

**Comment by bfriel** [comment]
@fakenine Solana Foundation recently announced an [RFP](https://forum.solana.com/t/the-solana-app-for-ledger-devices/367) for adding `signMessage` and other features to the Solana Ledger App. I've heard that they've made progress on implementing `signMessage` specifically, but I have not heard back on an ETA. Phantom will be ready to support this as soon as the Ledger app is ready for release.

**Comment by n526** [comment]
> @fakenine Solana Foundation recently announced an [RFP](https://forum.solana.com/t/the-solana-app-for-ledger-devices/367) for adding `signMessage` and other features to the Solana Ledger App. I've heard that they've made progress on implementing `signMessage` specifically, but I have not heard back on an ETA. Phantom will be ready to support this as soon as the Ledger app is ready for release.

Just in time!! Thanks

**Comment by c4b4d4** [comment]
What's the update on this? It's really needed in the ecosystem, and we've been asking it for a while

**Comment by c4b4d4** [comment]
Seems that the ETA is by the end of the year https://forum.solana.com/t/the-solana-app-for-ledger-devices/367/3 🚀

**Comment by mindrunner** [comment]
I wonder if there has been any progress. It seems to be still an issue for me.

**Comment by MHHukiewitz** [comment]
Bumping this, cause it still does not seem resolved, even though Ledger Solana firmware v1.5.1 is out :(
